### PR TITLE
[FIX] mass_mailing: Link `mail.mass_mailing.campaign` to `utm.campaign` records

### PR DESCRIPTION
**Pull request by yajo**
Method docstring explains the change.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr
@Tecnativa

### CODE REVIEW

**Review by rafaelbn — APPROVED** [review]
Tested functionally in production

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Do it on post-migration, as models are still not registered and you are using ORM.

File: addons/mass_mailing/migrations/10.0.2.0/pre-migration.py [hunk 1]
```diff
@@ -32,7 +32,32 @@ def fill_mass_mailing_sources(env):
...
+    """
+    mass = env["mail.mass_mailing.campaign"].search([
+        "|", ("campaign_id", "=", False), ("campaign_id.name", "=", False),
+    ])
+    for one in mass:
+        env.cr.execute(
+            "SELECT name FROM mail_mass_mailing_campaign WHERE id = %s",
+            (one.id,)
+        )
+        name = env.cr.fetchone()[0]
```
Comment (pedrobaeza) [inline addons/mass_mailing/migrations/10.0.2.0/pre-migration.py]: Please protect this from non results:
```
row = env.cr.fetchone()
if not row:
    continue
...
```

Comment (StefanRijnhart) [reply addons/mass_mailing/migrations/10.0.2.0/pre-migration.py]: @pedrobaeza given the previous lines it seems that each id is guaranteed to exist in the database.

Comment (yajo) [reply addons/mass_mailing/migrations/10.0.2.0/pre-migration.py]: Indeed, the failure is not possible here.

**Comment by yajo** [comment]
Thanks for the tips, let's see now

**Review by StefanRijnhart — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]
Please squash before merging

**Pull request merged**